src/docs/arch-storage-extension.dox

```diff
@@ -0,0 +1,104 @@
+/*! @arch_page arch-cloud-storage-extension Storage Source
+
+@section extension_storage_source Cloud Storage Source Implementation
+
+The WiredTiger extension allows the control of how files are stored via the storage source
+abstraction layer. More information can be found through @ref custom_storage_sources. There are
+three storage sources that extends the WiredTiger's capability to store the data files onto the
+cloud storage through Azure, Google and AWS providers. The cloud storage extensions provides a
+means to construct, configure and control a custom file system linked to a cloud bucket. All three
+storage source implementations internally interacts with the cloud storage through their own C++
+SDKs.
+
+All storage sources implements the following key functionalities defined by the storage source 
+interface:
+- Initialize the store, the AWS SDK and create a logging module for the store and the SDK.
+
+- Provide an API to access and terminate the store. A reference count is incremented each time the
+storage source reference is taken and decremented with the termination. When the reference count
+goes zero, the storage source is de-registered with WiredTiger and any allocated resources are
+freed.
+
+- Another essential functionality is that the storage source provides is to create a
+\c WT_FILE_SYSTEM implementation for a given bucket. The store requires a bucket and an access token
+to create the filesystem.
+
+@section extension_connection Connection class
+
+All extensions has a connection class which represents an active connection to a bucket. It
+encapsulates and implements all the operations the filesystem requires. The instantiation of the 
+class object authenticates with the the cloud provider and validates the existence of and access to
+the configured bucket. The connection class exposes an API to list the bucket contents filtered by a
+directory and a prefix, check for an object's existence in the bucket, put an object to the cloud,
+and get the object from the cloud. Though not required for the file system's implementation, the
+class also provides the means to delete the objects to clean up artifacts from the internal unit
+testing.
+
+@section extension_file_system FILE_SYSTEM
+
+WiredTiger supports a database over a custom file system by giving a \c WT_FILE_SYSTEM interface for
+the end-user to implement. The store itself provides a means to configure as many file systems as
+needed, each linked to a bucket. These file systems, called the \c GCP_FILE_SYSTEM, AZURE_FILE_SYSTEM 
+and S3_FILE_SYSTEM, are a custom implementation provided by the storage source. The filesystems 
+contains an instance to the connection class, hence providing an active connection to the bucket 
```

Comment:
Grammar: "The filesystems contains an instance to the connection class" -> "The filesystems contain an instance of the connection class"